pkg/pillar/scripts/device-steps.sh

```diff
@@ -121,9 +139,21 @@ fi
 
 CONFIGDEV=$(zboot partdev CONFIG)
 
+# If zedbox is already running we don't have to start it.
+if ! pgrep zedbox >/dev/null; then
+    echo "$(date -Ins -u) Starting zedbox"
+    $BINDIR/zedbox &
+    wait_for_touch zedbox
+fi
+
+mkdir -p "$WATCHDOG_PID" "$WATCHDOG_FILE"
+touch "$WATCHDOG_PID/zedbox.pid" "$WATCHDOG_FILE/zedbox.touch"
+
 if [ -c $TPM_DEVICE_PATH ] && ! [ -f $CONFIGDIR/disable-tpm ]; then
 #It is a device with TPM, enable disk encryption
-    if ! $BINDIR/vaultmgr setupVaults; then
+    $BINDIR/start-service.sh vaultmgr setupVaults &
```

Comment:
Another option here (if we can't wrap it into proper pubsub) would be to wait on the socket to reply with "done". That way -- one shot services like vaultmgr, etc. will reply "done" on the socket when they are done and everything else will reply immeditately.

start-service.sh will then simply block on reading from the socket and all the complexity will be gone.

Reply:
> Another option here (if we can't wrap it into proper pubsub) would be to wait on the socket to reply with "done". That way -- one shot services like vaultmgr, etc. will reply "done" on the socket when they are done and everything else will reply immeditately.

We'd still need to change device-steps.sh if this approach unless we want the zedagent etc start processes to sit around in the background and wait for the done forever; we need to get the start to complete and not consume memory.